.circleci/config.yml

```diff
@@ -31,18 +31,22 @@ jobs:
       - run:
           name: Compile test contracts
           command: ./scripts/compile_contracts.sh
+      - run:
+          name: Check out previous test metadata
+          command: | 
+            cat "${CIRCLE_INTERNAL_TASK_DATA}/circle-test-results/results.json" | jq .
       - run:
           name: Run pytest tests
           parallel: true
           command: |
-            TESTFILES=$(
-                circleci tests glob test/test*.py |
-                circleci tests split --split-by=timings --timings-type=classname
-            )
             mkdir -p test-results
-            poetry run pytest -s --junitxml=test-results/junit.xml -vv $TESTFILES
+            SPLITED_FILES_CLASS=$(circleci tests glob test/test*.py | sed  "s/\.py//g; s/\//./g" | circleci tests split --split-by=timings --timings-type=classname)
```

Comment:
- Why does this `sed` have two replace scripts, and the one in the next line has three? Is it because SPLITED_FILES_CLASS is a one-line string?
- Why are you using SPLITED_FILES_CLASS as an intermediary variable? Can't the output of `circleci test split` be piped into `sed`? Then maybe the next line could have just two replace scripts.
- Just in case, can you change `s/\.py//g` to `s/\.py$//g` so that we're sure that it only alters the extension.

Reply:
- Yes
- It can, but with an intermediate variable, it is more readable what's going on. It still need to have three, one to restore / one to retore .py to all words 'except last one because it has no space' and one to replace to last one.
- It will not work because only the last word will receive an extension. This is one line.